gobblin-core/src/main/java/gobblin/writer/AvroDataWriterBuilder.java

```diff
@@ -32,7 +32,11 @@
  * @author ynli
  */
 @SuppressWarnings("unused")
-public class AvroDataWriterBuilder extends DataWriterBuilder<Schema, GenericRecord> {
+public class AvroDataWriterBuilder extends PartitionAwareDataWriterBuilder<Schema, GenericRecord> {
+
+  @Override public boolean validatePartitionSchema(Schema partitionSchema) {
```

Comment:
Is this a good idea? I don't think that most partition aware writers will be able to handle all schemas, so I prefer to require implementers to explicitly think about what their writer will support.